doc/rst/meta/compiler-internals-doxygen/queries-impl.rst

```diff
@@ -0,0 +1,221 @@
+.. default-domain:: cpp
+
+.. _Chapter-dyno-chpl-queries-impl:
+
+Depths of the Query System
+==========================
+
+Vocabulary
+----------
+
+-  **Dependency**: a query executed as part of another query. The
+   “outer” query is said to depend on the “inner” query.
```

Comment:
"outer" and "inner" do not make sense to me in this context. Might it say, "Suppose a query A runs another query B. In that case, A is said to depend on B."